app/controllers/template_invocations_controller.rb

```diff
@@ -17,4 +22,56 @@ def show
     @line_sets = @line_sets.drop_while { |o| o['timestamp'].to_f <= @since } if @since
     @line_counter = params[:line_counter].to_i
   end
+
+  def show_template_invocation_by_host
+    @template_invocation = @job_invocation.template_invocations.find { |template_inv| template_inv.host_id == @host.id }
+    if @template_invocation.nil?
+      render :json => { :error => _('Template invocation not found') }, :status => :not_found
+    end
+    @template_invocation_task = @template_invocation.run_host_job_task
+
+    lines = normalize_line_sets(@template_invocation_task.main_action.live_output)
+    transformed_input_values = @template_invocation.input_values.joins(:template_input).map do |input_value|
+      {
+        name: input_value&.template_input&.name,
+        value: input_safe_value(input_value),
+      }
+    end
+
+    auto_refresh = @job_invocation.task.try(:pending?)
+    finished = @job_invocation.status_label == 'failed' || @job_invocation.status_label == 'succeeded' || @job_invocation.status_label == 'cancelled'
+    render :json => { :output => lines, :preview => template_invocation_preview(@template_invocation, @host), :input_values => transformed_input_values, :job_invocation_description => @job_invocation.description, :task_id => @template_invocation_task.id, :task_cancellable => @template_invocation_task.cancellable?, :host_name => @host.name, :permissions => {
+      :view_foreman_tasks => User.current.allowed_to?(:view_foreman_tasks),
+      :cancel_job_invocations => User.current.allowed_to?(:cancel_job_invocations),
+      :execute_jobs => User.current.allowed_to?(:create_job_invocations) && (!@host.infrastructure_host? || User.current.can?(:execute_jobs_on_infrastructure_hosts)),
+
+    },
+    :auto_refresh => auto_refresh, :finished => finished}, status: :ok
+  end
+
+  private
+
+  def find_job_invocation
+    @job_invocation = JobInvocation.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    render :json => { :error => { :message => format(_("Job with id '%{id}' was not found"), :id => params['id']) } }, :status => :not_found
+  end
+
+  def find_host
+    @host = Host.find(params[:host_id])
+  rescue ActiveRecord::RecordNotFound
+    render :json => { :error => { :message => format(_("Host with id '%{id}' was not found"), :id => params['host_id']) } }, :status => :not_found
```

Comment:
There should be a generic handler already present that should catch ActiveRecord::RecordNotFound and render an appropriate response. Wouldn't that work?

Reply:
Using `not_found({ :error => { :message => (_("Host with id '%{id}' was not found") % { :id => params['host_id'] }) } })` doesnt show the error in the response but just gives 404, since a few different things can be not found I thought it will be best to specifiy